### ISSUE TITLE
stage JS error for db_LB

### ISSUE THREAD

**Issue by ozten** [mozilla/persona#1057]
STR
1) Set language in browser to it-CH
2) Go to http://beta.myfavoritebeer.org/
3) Click Sign in

Expected:
Sign in flow

Actual:
Dialog stuck on Communicating with server.
Note JS Error in console

Error: missing ) after argument list
Source File: https://diresworb.org/production/db_LB/dialog.js
Line: 9, Column: 20692
Source Code:
:function(a,b){var c=this;return function(d){c.renderError("error",$.extend({action:a},d),b)}},onSubmit:h});return k}(),BrowserID.Modules.XHRDelay=function(){function f(){c&&(c=!1,this.hideDelay())}function e(){c=!0,this.renderDelay("wait",b.slowXHR)}"use strict";var a=BrowserID,b=a.Wait,c,d,g=a.Modules.PageModule.extend({start:function(a){var b=this;b.subscribe("xhr_delay",e),b.subscribe("xhr_complete",f),d.start.call(b,a)},stop:function(){this.hideDelay(),d.stop.call(this)}});d=g.sc;return g}(),BrowserID.Modules.XHRDisableForm=function(){"use strict";var a=BrowserID,b=a.DOM,c,d=a.Modules.PageModule.extend({start:function(a){var d=this;d.subscribe("xhr_start",b.addClass.curry("body","submit_disabled")),d.subscribe("xhr_complete",b.removeClass.curry("body","submit_disabled")),c.start.call(d,a)}});c=d.sc;return d}(),BrowserID.Modules.CodeCheck=function(){function h(a,b){var c=document.createElement("script");c.type="text/javascript",c.src="https://diresworb.org/production/"+e+"_v"+a+".js",document.head.appendChild(c),b()}function g(a,b){function e(){a&&a(c)}var c=b===d;c?e():h(b,e)}function f(b){a.Network.codeVersion(b,this.getErrorDialog(a.Errors.checkScriptVersion,b))}"use strict";var a=BrowserID,b=a.DOM,c,d,e,i=a.Modules.PageModule.extend({start:function(a){function h(b){a.ready&&a.ready(b)}var b=this;a=a||{},b.checkRequired(a,"code_ver","file_name_prefix"),d=a.code_ver,e=a.file_name_prefix,f.call(b,function(a){a?g.call(b,h,a):h(!0)}),c.start.call(b,a)}});c=i.sc;return i}(),BrowserID.Modules.CookieCheck=function(){"use strict";var a=BrowserID,b=a.Helpers.complete,c=a.Network,d=a.Errors,e,f=a.Modules.PageModule.extend({start:function(a){var f=this;c.cookiesEnabled(function(c){c||f.renderError("generic",d.cookiesDisabled),b(a.ready,c)},f.getErrorDialog(d.cookiesEnabled,a.ready)),e.start.call(f,a)}});e=f.sc;return f}(),function(){function g(a,b,c){function e(a){c&&c(a)}d.checkAuthenticationAndSync(function(c){c?(d.setOrigin(a),d.getAssertion(b,d.getOrigin(),function(a){e(a||null)},e.curry(null))):e(null)},e.curry(null))}"use strict";var a=navigator,b=BrowserID,c=b.internal=b.internal||{},d=b.User,e=b.Storage,f=b.module;c.setPersistent=function(a,b){function c(a){b&&b(a)}d.checkAuthentication(function(b){b&&e.site.set(a,"remember",!0),c(!!b||null)},c.curry(null))},c.get=function(a,b,c){function d(a){b&&b(a||null)}c=c||{};var h=!!c.silent;if(h){var i=c.requiredEmail||e.site.get(a,"email");i?g(a,i,b):d()}else{var j=f.getRunningModule("dialog");j?j.get(a,c,d,d):d()}}}(),function(){function l(a,b){var g=this;d.getStoredEmailKeypair(a)?(e.showTooltip("#already_own_address"),c(b,!1)):d.addressInfo(a,function(h){if(h.type==="primary"){var h=_.extend(h,{email:a,add:!0});g.close("primary_user",h,h),c(b,!0)}else d.addEmail(a,function(d){if(d){var f={email:a};g.close("email_staged",f,f)}else e.showTooltip("#could_not_add");c(b,d)},g.getErrorDialog(f.addEmail,b))},g.getErrorDialog(f.addressInfo,b))}function k(a,b){var g=this;d.requestPasswordReset(a,function(d){d.success?g.close("reset_password",{email:a}):e.showTooltip("#could_not_add"),c(b,d.success)},g.getErrorDialog(f.requestPasswordReset,b))}function j(a,b){var g=this;d.createSecondaryUser(a,function(d){if(d){var f={email:a};g.close("user_staged",f,f),c(b,!0)}else e.showTooltip("#could_not_add"),c(b,!1)},g.getErrorDialog(f.createUser,b))}function i(a,b,g){var h=this;d.authenticate(a,b,function(a){a||e.showTooltip("#cannot_authenticate"),c(g,a)},h.getErrorDialog(f.authenticate,g))}function h(b,e){var h=this,i=a.Screens.wait;i.show("wait",a.Wait.generateKey),d.getAssertion(b,d.getOrigin(),function(a){a=a||null,i.hide(),g(function(){h.close("assertion_generated",{assertion:a}),c(e,a)})},h.getErrorDialog(f.getAssertion,c))}function g(a){var b=$("body"),d=$("#signIn").length&&b.innerWidth()>640;d?($("#signIn").animate({width:"95%"},750,function(){b.delay(500).animate({opacity:"0.5"},500)}),setTimeout(c.curry(a),1750)):c(a)}"use strict";var a=BrowserID,b=a.Helpers,c=b.complete,d=a.User,e=a.Tooltip,f=a.Errors;b.Dialog=b.Dialog||{},b.extend(b.Dialog,{getAssertion:h,authenticateUser:i,createUser:j,addEmail:l,resetPassword:k,cancelEvent:b.cancelEvent})}(),BrowserID.State=function(){function k(){var c=this,k=c.subscribe.bind(c),l=function(a,b,d){typeof a!="boolean"&&(d=b,b=a,a=!0);var e=g[b].bind(g);c.gotoState(a,e,d)};cancelState=c.popState.bind(c),k("offline",function(a,b){l("doOffline")}),k("start",function(b,d){d=d||{},c.hostname=d.hostname,c.allowPersistent=!!d.allowPersistent,j=d.requiredEmail,typeof j!="undefined"&&!a.verifyEmail(j)?l("doError","invalid_required_email",{email:j}):l("doCheckAuth")}),k("cancel",function(){l("doCancel")}),k("window_unload",function(){c.success||(a.Storage.setStagedOnBehalfOf(""),l("doCancel"))}),k("authentication_checked",function(a,b){var c=b.authenticated;j?l("doAuthenticateWithRequiredEmail",{email:j}):c?d("pick_email"):d("authenticate")}),k("authenticate",function(a,b){l("doAuthenticate",b)}),k("user_staged",function(a,b){c.stagedEmail=b.email,b.required=!!j,l("doConfirmUser",b)}),k("user_confirmed",function(){l("doEmailConfirmed",{email:c.stagedEmail})}),k("primary_user",function(a,c){h=!!c.add,i=c.email;var e=b.getEmail(i);e&&e.cert?d("primary_user_ready",c):l(!1,"doProvisionPrimaryUser",c)}),k("primary_user_provisioned",function(a,b){b=b||{},b.add=!!h,l("doPrimaryUserProvisioned",b)}),k("primary_user_unauthenticated",function(a,b){b=b||{},b.add=!!h,b.email=i,b.requiredEmail=!!j,l("doVerifyPrimaryUser",b)}),k("primary_user_authenticating",function(a,b){f.stopAll(),c.success=!0}),k("primary_user_ready",function(a,b){l("doEmailChosen",b)}),k("pick_email",function(){l("doPickEmail",{origin:c.hostname,allow_persistent:c.allowPersistent})}),k("email_chosen",function(a,c){function g(){c.complete&&c.complete()}var f=c.email;idInfo=b.getEmail(f);if(idInfo)idInfo.type==="primary"?idInfo.cert?l("doEmailChosen",c):d("primary_user",c):e.checkAuthentication(function(a){a==="assertion"?l("doAuthenticateWithRequiredEmail",{email:f,secondary_auth:!0}):l("doEmailChosen",c),g()},g);else throw"invalid email"}),k("notme",function(){l("doNotMe")}),k("logged_out",function(){d("authenticate")}),k("authenticated",function(a,b){d("pick_email")}),k("forgot_password",function(a,b){l(!1,"doForgotPassword",b)}),k("reset_password",function(a,b){l(!1,"doResetPassword",b)}),k("assertion_generated",function(a,b){c.success=!0,b.assertion!==null?l("doAssertionGenerated",b.assertion):l("doPickEmail")}),k("add_email",function(a,b){l("doAddEmail")}),k("email_staged",function(a,b){c.stagedEmail=b.email,b.required=!!j,l("doConfirmEmail",b)}),k("email_confirmed",function(){l("doEmailConfirmed",{email:c.stagedEmail})}),k("cancel_state",function(a,b){cancelState(b)})}var a=BrowserID,b=a.Storage,c=a.Mediator,d=c.publish.bind(c),e=a.User,f=a.module,g,h=!1,i,j,l=BrowserID.StateMachine.extend({start:function(a){a=a||{},g=a.controller;if(!g)throw"start: controller must be specified";h=!1,i=j=null,l.sc.start.call(this,a),k.call(this)}});return l}(),BrowserID.Modules.Actions=function(){function k(a,b,c){var d=j("check_registration",{email:a.email,required:a.required,verifier:b,verificationMessage:c});d.startCheck()}function j(a,b){g&&c.stop(g);var d=c.start(a,b);d&&(g=a);return d}"use strict";var a=BrowserID,b,c=a.module,d=a.User,e=a.Errors,f=a.Wait,g,h,i,l=a.Modules.PageModule.extend({start:function(a){var c=this;a=a||{},h=a.onsuccess,i=a.onerror,b.start.call(c,a),a.ready&&_.defer(a.ready)},doError:function(a,b){b||(b=a,a="error"),this.renderError(a,b)},doOffline:function(){this.renderError("offline",{})},doCancel:function(){h&&h(null)},doConfirmUser:function(a){k.call(this,a,"waitForUserValidation","user_confirmed")},doPickEmail:function(a){j("pick_email",a)},doAddEmail:function(a){j("add_email",a)},doAuthenticate:function(a){j("authenticate",a)},doAuthenticateWithRequiredEmail:function(a){j("required_email",a)},doForgotPassword:function(a){j("forgot_password",a)},doResetPassword:function(a){this.doConfirmUser(a.email)},doConfirmEmail:function(a){k.call(this,a,"waitForEmailValidation","email_confirmed")},doEmailConfirmed:function(a){var b=this;d.getAssertion(a.email,d.getOrigin(),function(a){b.publish("assertion_generated",{assertion:a})},b.getErrorDialog(e.getAssertion))},doAssertionGenerated:function(a){i=null,h&&h(a)},doNotMe:function(){var a=this;d.logoutUser(a.publish.bind(a,"logged_out"),a.getErrorDialog(e.logoutUser))},doCheckAuth:function(){var a=this;d.checkAuthenticationAndSync(function(b){a.publish("authentication_checked",{authenticated:b})},a.getErrorDialog(e.checkAuthentication))},doProvisionPrimaryUser:function(a){j("provision_primary_user",a)},doVerifyPrimaryUser:function(a){j("verify_primary_user",a)},doPrimaryUserProvisioned:function(a){j("primary_user_provisioned",a)},doEmailChosen:function(a){j("email_chosen",a)}});b=l.sc;return l}(),BrowserID.Modules.Dialog=function(){function n(){this.publish("window_unload")}function m(a){b.setOrigin(a),d.setInner("#sitename",b.getHostname())}function l(){f&&f.detach()}function k(){var a=this,b=e.location.hash;if(e.navigator.id&&e.navigator.id.channel)e.navigator.id.channel.registerController(a);else{if(b=="#NATIVE"||b=="#INTERNAL")return;try{f=WinChan.onOpen(function(b,c,d){c&&a.get(b,c.params,function(a){d(a)},function(a){d(null)})})}catch(d){a.renderError("error",{action:c.relaySetup})}}}function j(a){var b=BrowserID.State.create();b.start({controller:a})}function i(a,b){var c=BrowserID.Modules.Actions.create();c.start({onsuccess:a,onerror:b});return c}function h(){return true}"use strict";var a=BrowserID,b=a.User,c=a.Errors,d=a.DOM,e=window,f,g,o=a.Modules.PageModule.extend({start:function(a){var b=this;a=a||{},e=a.window||window,g.start.call(b,a),k.call(b),a.ready&&_.defer(a.ready)},stop:function(){l(),g.stop.call(this)},getVerifiedEmail:function(a,b,c){return this.get(a,{},b,c)},get:function(a,c,d,f){var g=this,k=e.location.hash;m(a);var l=i.call(g,d,f);j.call(g,l);if(h.call(g)){c=c||{},c.hostname=b.getHostname(),g.bind(e,"unload",n);if(k.indexOf("#CREATE_EMAIL=")===0){var o=k.replace(/#CREATE_EMAIL=/,"");g.renderDialog("primary_user_verified",{email:o}),g.close("primary_user",{email:o,add:!1})}else if(k.indexOf("#ADD_EMAIL=")===0){var o=k.replace(/#ADD_EMAIL=/,"");g.renderDialog("primary_user_verified",{email:o}),g.close("primary_user",{email:o,add:!0})}else g.publish("start",c)}},onWindowUnload:n});g=o.sc;return o}(),BrowserID.Modules.Authenticate=function(){function x(){var a=k.getInner("#email");a!==l&&(l=a,t.call(this))}function w(){var a=this;a.publish("create_user"),a.submit=q,s(".start:visible,.returning:visible",".newuser")}function v(){var a=n();if(a){var b=m||{email:a};this.close("forgot_password",b,b)}}function u(a){var b=this;b.publish("enter_password",m),b.submit=r,s(".start:visible,.newuser:visible,.forgot:visible",".returning",function(){k.focus("#password")}),j(a)}function t(a){$("#email").is(":disabled")||(this.submit=p,s(".returning:visible,.newuser:visible",".start"))}function s(b,c,d){$(b).hide(),$(c).fadeIn(a,d)}function r(){var a=n(),b=g.getAndValidatePassword("#password"),c=this;a&&b&&h.authenticateUser.call(c,a,b,function(b){b&&c.close("authenticated",{email:a})})}function q(a){var b=this,c=n();c?h.createUser.call(b,c,a):a&&a()}function p(a){function f(a){m=a,a.type==="primary"?e.close("primary_user",a,a):a.known?u.call(e):w.call(e)}var b=n(),e=this;!b||(a&&a.type?f(a):c.addressInfo(b,f,e.getErrorDialog(d.addressInfo)))}function o(a){var b=this;b.submit=p,a&&a.email&&a.type==="secondary"&&a.known?u.call(b,a.ready):(s(".newuser,.forgot,.returning",".start"),j(a.ready))}function n(){return g.getAndValidateEmail("#email")}"use strict";var a=250,b=BrowserID,c=b.User,d=b.Errors,e=b.Validation,f=b.Tooltip,g=b.Helpers,h=g.Dialog,i=g.cancelEvent,j=g.complete,k=b.DOM,l="",m,y=b.Modules.PageModule.extend({start:function(a){a=a||{},m=null,l=a.email||"";var b=this;b.renderDialog("authenticate",{sitename:c.getHostname(),email:l}),$(".newuser,.forgot,.returning,.start").hide(),b.bind("#email","keyup",x),b.bind("#forgotPassword","click",i(v)),y.sc.start.call(b,a),o.call(b,a)},checkEmail:p,createUser:q,authenticate:r,forgotPassword:v});return y}(),BrowserID.Modules.ForgotPassword=function(){function h(){this.close("cancel_state",{email:this.email})}function g(){var a=this;d.resetPassword.call(a,a.email)}"use strict";var a=250,b=BrowserID,c=b.Helpers,d=c.Dialog,e=d.cancelEvent,f=b.DOM,i=b.Modules.PageModule.extend({start:function(a){var b=this;b.email=a.email,b.renderDialog("forgot_password",{email:a.email||"",requiredEmail:a.requiredEmail}),b.bind("#cancel","click",e(h)),i.sc.start.call(b,a)},submit:g,resetPassword:g,cancelResetPassword:h});return i}(),BrowserID.Modules.CheckRegistration=function(){"use strict";var a=BrowserID,b=a.User,c=a.DOM,d=a.Errors,e=a.Modules.PageModule.extend({start:function(a){var b=this;a=a||{},a.required=!!a.required,b.renderWait("confirm_email",a),b.email=a.email,b.verifier=a.verifier,b.verificationMessage=a.verificationMessage,b.bind("#back","click",b.back),b.bind("#cancel","click",b.cancel),e.sc.start.call(b,a)},startCheck:function(a){var c=this;b[c.verifier](c.email,function%28d%29{d==="complete"?b.syncEmails%28function%28%29{c.close%28c.verificationMessage%29,a&&a%28%29}%29:d==="mustAuth"&&%28c.close%28"authenticate",{email:c.email}%29,a&&a%28%29%29},c.getErrorDialog%28d.registration,a%29)},back:function(){b.cancelUserValidation(),this.close("cancel_state")},cancel:function(){this.close("cancel")}});return e}(),BrowserID.Modules.PickEmail=function(){function m(a){i(a);var c=this,e=g.getInner("input[type=radio]:checked"),f=l.call(c,e);if(f){var h=b.getOrigin();d.site.set(h,"email",e),c.allowPersistent&&d.site.set(h,"remember",$("#remember").is(":checked")),c.close("email_chosen",{email:e})}}function l(a){var c=b.getStoredEmailKeypair(a);c||(alert("The selected email is invalid or has been deleted."),this.close("assertion_generated",{assertion:null}));return!!c}function k(a){i(a),this.close("add_email")}function j(a){i(a);var b=this;g.getElements("input[type=radio]:checked").length||g.setAttr("input[type=radio]:eq(0)","checked",!0),g.focus("input[type=radio]:checked"),b.submit=m}function i(a){a&&a.preventDefault()}"use strict";var a=BrowserID,b=a.User,c=a.Errors,d=a.Storage,e=a.Helpers,f=e.Dialog,g=a.DOM,h,n=a.Modules.PageModule.extend({start:function(a){var c=b.getOrigin(),e=this;a=a||{},e.allowPersistent=a.allow_persistent,g.addClass("body","pickemail"),e.renderDialog("pick_email",{identities:b.getStoredEmailKeypairs(),siteemail:d.site.get(c,"email"),allow_persistent:a.allow_persistent||!1,remember:d.site.get(c,"remember")||!1}),g.getElements("body").css("opacity","1"),g.getElements("#selectEmail input[type=radio]:visible").length===0&&g.focus("#signInButton"),e.bind("#useNewEmail","click",k),h.start.call(e,a),j.call(e)},stop:function(){h.stop.call(this),g.removeClass("body","pickemail")},signIn:m,addEmail:k});h=n.sc;return n}(),BrowserID.Modules.AddEmail=function(){function i(){this.close("cancel_state")}function h(a){var d=b.getAndValidateEmail("#newEmail"),e=this;d?c.addEmail.call(e,d,a):f(a,!1)}"use strict";var a=BrowserID,b=a.Helpers,c=b.Dialog,d=c.cancelEvent,e=a.Errors,f=b.complete,g=a.Tooltip,j=a.Modules.PageModule.extend({start:function(a){var b=this;b.renderDialog("add_email",a),b.bind("#cancel","click",d(i)),j.sc.start.call(b,a)},submit:h,addEmail:h,cancelAddEmail:i});return j}(),BrowserID.Modules.RequiredEmail=function(){function r(){this.close(m?"cancel_state":"cancel")}function q(){var a=this;a.close("forgot_password",{email:j,requiredEmail:!0})}function p(){var a=this;k?f.addEmail.call(a,j):f.createUser.call(a,j)}function o(a){function c(){f.getAssertion.call(b,j,a)}var b=this;if(l)n.call(b,a);else if(k==="password")c();else{var d=e.getAndValidatePassword("#password");d&&f.authenticateUser.call(b,j,d,function(b){b?c():a&&a()})}}function n(a){this.close("primary_user",_.extend(l,{email:j,requiredEmail:!0,add:!!k})),a&&a()}"use strict";var a=250,b=BrowserID,c=b.User,d=b.Errors,e=b.Helpers,f=e.Dialog,g=b.DOM,h,i=f.cancelEvent,j,k,l,m,s=b.Modules.PageModule.extend({start:function(a){function f(a){a=_.extend({email:j,verify:!1,signin:!1,password:!1,secondary_auth:!1,primary:!1},a),b.renderDialog("required_email",a),b.bind("#sign_in","click",i(o)),b.bind("#verify_address","click",i(p)),b.bind("#forgotPassword","click",i(q)),b.bind("#cancel","click",i(r))}function e(){a.ready&&a.ready()}var b=this;j=a.email||"",m=a.secondary_auth,l=null,c.checkAuthentication(function(a){k=a;var g=c.getStoredEmailKeypair(j);g&&g.type==="secondary"?(f({signin:!0,password:k!=="password",secondary_auth:m}),e()):g&&g.cert?(l=g,f({signin:!0,primary:!0}),e()):c.addressInfo(j,function(a){a.type==="primary"&&(l=a),a.authed?f({signin:!0,primary:!0}):g||a.type==="primary"?n.call(b):k||!a.known?f({verify:!0}):f({signin:!0,password:!0}),e()},b.getErrorDialog(d.addressInfo,e))},b.getErrorDialog(d.checkAuthentication,e)),s.sc.start.call(b,a)},signIn:o,verifyAddress:p,forgotPassword:q,cancel:r});return s}(),BrowserID.Modules.VerifyPrimaryUser=function(){function k(a){this.close("cancel_state"),a&&a()}function j(a){this.publish("primary_user_authenticating");var b=c.document.location.href.replace(/#.*$/,""),i=d?"ADD_EMAIL":"CREATE_EMAIL",j=g.toURL(f,{email:e,return_to:b+"#"+i+"="+e});c.document.location=j,h(a)}"use strict";var a=BrowserID,b,c,d,e,f,g=a.Helpers,h=g.complete,i=g.Dialog.cancelEvent,l=a.Modules.PageModule.extend({start:function(a){var g=this;a=a||{},c=a.window||window,d=a.add,e=a.email,f=a.auth_url,a.requiredEmail=a.requiredEmail||!1,g.renderDialog("verify_primary_user",a),g.bind("#cancel","click",i(k)),b.start.call(g,a)},submit:j,cancel:k});b=l.sc;return l}(),BrowserID.Modules.ProvisionPrimaryUser=function(){function e(a,b,e,f){function h(a){f&&f(a)}var g=this;c.provisionPrimaryUser(a,{auth:b,prov:e},function(c,d){switch(c){case"primary.already_added":break;case"primary.verified":g.close("primary_user_provisioned",{email:a,assertion:d.assertion}),h(!0);break;case"primary.verify":g.close("primary_user_unauthenticated",{email:a,auth_url:b}),h(!0);break;case"primary.could_not_add":break;default:}},g.getErrorDialog(d.provisioningPrimary))}"use strict";var a=250,b=BrowserID,c=b.User,d=b.Errors,f=b.Modules.PageModule.extend({start:function(a){a=a||{};var b=this,g=a.email,h=a.auth,i=a.prov;if(!g)throw"missing config option: email";!h||!i?c.addressInfo(g,function(a){a.type==="primary"?e.call(b,g,a.auth,a.prov):b.renderError("error",{action:d.provisioningBadPrimary})},b.getErrorDialog(d.isEmailRegistered)):e.call(b,g,h,i),f.sc.start.call(b,a)},provisionPrimaryUser:e});return f}(),BrowserID.Modules.PrimaryUserProvisioned=function(){"use strict";var a=250,b=BrowserID,c=b.User,d=b.Network,e=b.Errors,f=b.Modules.PageModule.extend({start:function(a){a=a||{};var b=this,c=a.email,g=a.assertion,h=!!a.add,i=function(b){a.ready&&a.ready(b||!1)},j=a.complete_delay||3e3;b.checkRequired(a,"email","assertion"),b.renderDialog("primary_user_verified",{email:c}),h?d.addEmailWithAssertion(g,function(c){c?setTimeout(function(){b.publish("primary_user_ready",a)},j):b.getErrorDialog(e.addEmailWithAssertion,i)()},b.getErrorDialog(e.addEmailWithAssertion,i)):d.authenticateWithAssertion(c,g,function(c){c?setTimeout(function(){b.publish("primary_user_ready",a)},j):b.getErrorDialog(e.authenticateWithAssertion,i)()},b.getErrorDialog(e.authenticateWithAssertion,i)),f.sc.start.call(b,a)}});return f}(),BrowserID.Modules.EmailChosen=function(){"use strict";var a=BrowserID,b=a.Helpers.Dialog,c,d=a.Modules.PageModule.extend({start:function(a){var d=a.email,e=this;if(!d)throw"email required";b.getAssertion.call(e,d,a.ready),c.start.call(e,a)}});c=d.sc;return d}(),function(){var a=BrowserID,b=a.module,c=a.Modules,d=a.Network,e=a.XHR;e.init({time_until_delay:1e4}),d.init(),b.register("cookie_check",c.CookieCheck),b.start("cookie_check",{ready:function(a){!a||(b.register("code_check",c.CodeCheck),b.start("code_check",{file_name_prefix:"dialog",code_ver:"__BROWSERID_CODE_VERSION__",ready:function(a){a&&(b.register("dialog",c.Dialog),b.register("add_email",c.AddEmail),b.register("authenticate",c.Authenticate),b.register("check_registration",c.CheckRegistration),b.register("forgot_password",c.ForgotPassword),b.register("pick_email",c.PickEmail),b.register("required_email",c.RequiredEmail),b.register("verify_primary_user",c.VerifyPrimaryUser),b.register("provision_primary_user",c.ProvisionPrimaryUser),b.register("primary_user_provisioned",c.PrimaryUserProvisioned),b.register("email_chosen",c.EmailChosen),b.register("xhr_delay",c.XHRDelay),b.register("xhr_disable_form",c.XHRDisableForm),b.start("xhr_delay"),b.start("xhr_disable_form"),b.start("dialog"))}}))}}

**Comment by fetep** [comment]
Things I've observed so far:
- the file is being built properly (on-disk, it ends with "()")
- even bypassing zeus & nginx in stage, the file is being served without a "()" at the end.

Is some middleware breaking it, maybe?

**Comment by ozten** [comment]
Sorry for massive copy/paste.

More info...

node in stage is cutting off the end of dialog.js for the following locales:
- db_LB
- el
- ru

The larges JS files in order are: db_LB, sr, el, ru

It's not clear why sr works but the other three are truncated.

@fetep requested directly from node, to rule out nginx or zeus.

It's possible that a node is getting confused by the file encoding and serving up less bytes that the entire file.... Working on a way to test that hypothesis.

Node does not have an issue in local dev or https://temporary_preview.hacksign.in/production/db_LB/dialog.js

**Comment by ozten** [comment]
@petef what is the md5sum of `resources/static/production/db_LB/dialog.js` on stage? Can you attach the original file to this bug (since a wget via http will be hozed)

**Comment by ozten** [comment]
The following po files are detected as ASCII by `file`:
- en_US
- eo
- es_MX
- et
- eu
- fr
- gd
- gl
- hr
- ja
- lt
- ml
- pt_BR
- pt
- sk
- son

which the others being detected as UTF-8

**Comment by ozten** [comment]
Reference codebase is:

SVN locale @ r101191
Git repo/branch https://github.com/fetep/browserid/tree/train-2012.02.02-petef

**Comment by ozten** [comment]
HTTP header on response - Content-Length: 285467

`curl --ignore-header-length` causes complete JS file to be downloaded...

```
ls -l /tmp/db_LB/dialog.js 
-rw-r--r-- 1 ozten ozten 285471 2012-02-06 16:42 /tmp/db_LB/dialog.js
ozten@haskwhal:~/Projects/browserid$ tail -c50 /tmp/db_LB/dialog.js 
t("xhr_disable_form"),b.start("dialog"))}}))}})}()
```

**Comment by ozten** [comment]
Placing the offending file http://people.mozilla.com/~petef/db_LB_dialog.js on temporary_preview does not reproduce the issue.

**Comment by ozten** [comment]
@fetep here is a reduced test case https://gist.github.com/1761254

Can we put that on a browserid box and hit port 8666?

**Comment by ozten** [comment]
Via IRC (10:56:10 AM) petef: ozten: tha ttest case is running on web1.scl2.stage
(10:56:25 AM) petef: but now with curl --ignore-content-length, on both the test server and the real bid server, it's hanging after so many bytes
(10:56:34 AM) ***petef confused, will look again in a bit

**Comment by ozten** [comment]
@lloyd found https://gist.github.com/1772675

c9 a5 -> ef bf bd

which adds 4 bytes to the response.

**Comment by lloyd** [comment]
dirty.  stupid.  ugly.  bug.

tl;dr; bugs in postprocess

So!  concatenation of buffers using '+=' is not safe.  it does an implicit conversion to a javascript utf8 string.  if there happens to be a partial utf8 char that hangs over buffer boundaries, then the utf8 conversion translates the invalid utf8 into a REPLACEMENT CHAR or some such.  The result is munged chars.

The plot thickens.  Because postprocess checks to see if the content length of the buffer before substitution is different than the length after, and only then updates content-length, this issue would manifest when substitution string (i.e. `diresworb.org` is the same length as the original text `browserid.org`).  If that was not the case (i.e. `dev.diresworb.org`), we would update content length, including the bytes for the munged chars (which would add about 4 bytes, 2 for the munged char before the break, and 2 after). 

So that last paragraph explains why this problem seems to be reproducible in stage (replacement same length as orig), but not in ephemeral `hacksign.in` vms, (replacement length different).  what you _would_ be able to see in vms is munged text if you knew where to look.

summary: wrong content length was a symtpom of char munging, char munging is fixed by post-process `0.2.1`, oh happy day.

**Comment by lloyd** [comment]
ozten: this issue won't occur in production even if we don't merge the fix, but it will continue to plague us in the QA env.  

given that postprocess doesn't come into play in prod, I consider this very low risk.  would you like to hotfix?

**Comment by ozten** [comment]
This bug only affects 2 real locales and we feel pretty good about the behavior in production.

I'd say no need to hotfix, be ready to dive deeper if it does repro in prod.

After caching fix, we should look at removing postprocessing for hostnames, there may be a cachable runtime solution.
